### ISSUE TITLE
Attach MorphToMany with custom $parentKey

### ISSUE THREAD

**Issue by dododedodonl** [laravel/nova-issues#3493]
- Laravel Version: 8.53.0
- Nova Version: 3.28.0
- PHP Version: 8.0.9
- Database Driver & Version: mysql (mariadb 10.3.25)
- Operating System and Version: Arch Linux 2021-08-04 15:35:00 UTC
- Browser type and version: Firefox 90.0.2

### Description:
When using a non-default parent key on a morphToMany relation, nova tries to attach it using the `key` (resourceId) instead of the custom `parentKey` specified on the relation (sixth argument).

When the relationship is used directly (eg. `$firstModel->taggable()->attach($tag)`) it works as expected (`short_id` is used as key). When attaching something using Nova, `id` is used as key.

`\Laravel\Nova\Http\Controllers\MorphedResourceAttachController` extends `\Laravel\Nova\Http\Controllers\ResourceAttachController` and overrides `initializePivot`, where unlike its parent, just the resourceId is set.

### Detailed steps to reproduce the issue on a fresh Nova installation:
- create a model `FirstModel` with a `morphToMany` named `secondModel` with a custom `$parentKey` (sixth argument)
- create a model `Tag` to morph to
- create a resource for `FirstModel` with field `MorphToMany::make(__('Tag'), 'taggable')`
- try to attach a model

### Code
app/Models/FirstModel.php
```php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model

class FirstModel extends Model
{
    public function taggable(): MorphToMany
    {
        return $this->morphToMany(
            Tag::class,
            'taggable',
            null,
            null,
            null,
            'short_id',
        );
    }
}
```

app/Nova/FirstModel.php
```php
<?php

namespace App\Nova;

use Laravel\Nova\Fields\ID;
use Laravel\Nova\Fields\MorphToMany;

class FirstModel extends Resource
{
    public static $model = \App\Models\FirstModel::class;
    
    public function fields(Request $request)
    {
        return [
            ID::make()->sortable(),
            MorphToMany::make(__('Tag'), 'taggable'),
        ];
    }
}
```

**Comment by crynobone** [comment]
Can you create a working code repository showing the issue.

**Comment by dododedodonl** [comment]
https://github.com/dododedodonl/nova-morph-issue

I fixed it with this locally (inspired by `ResourceAttachController`):

```diff
diff --git a/src/Http/Controllers/MorphedResourceAttachController.php b/src/Http/Controllers/MorphedResourceAttachController.php
index 173cbb7..55af134 100644
--- a/src/Http/Controllers/MorphedResourceAttachController.php
+++ b/src/Http/Controllers/MorphedResourceAttachController.php
@@ -20,9 +20,23 @@ class MorphedResourceAttachController extends ResourceAttachController
             abort_unless($resource->hasRelatableField($request, $request->viaRelationship), 404);
         })->model();
 
+        $parentKey = $request->resourceId;
+        $relatedKey = $request->input($request->relatedResource);
+
+        $parentKeyName = $relationship->getParentKeyName();
+        $relatedKeyName = $relationship->getRelatedKeyName();
+
+        if ($parentKeyName !== $request->model()->getKeyName()) {
+            $parentKey = $request->findModelOrFail()->{$parentKeyName};
+        }
+
+        if ($relatedKeyName !== ($request->newRelatedResource()::newModel())->getKeyName()) {
+            $relatedKey = $request->findRelatedModelOrFail()->{$relatedKeyName};
+        }
+
         ($pivot = $relationship->newPivot())->forceFill([
-            $relationship->getForeignPivotKeyName() => $request->resourceId,
-            $relationship->getRelatedPivotKeyName() => $request->input($request->relatedResource),
+            $relationship->getForeignPivotKeyName() => $parentKey,
+            $relationship->getRelatedPivotKeyName() => $relatedKey,
             $relationship->getMorphType() => $model->{$request->viaRelationship}()->getMorphClass(),
         ]);
```

**Comment by crynobone** [comment]
This has been fixed with v3.29.0. https://nova.laravel.com/releases/3.29.0
